src/cascadia/UnitTests_SettingsModel/ProfileTests.cpp

```diff
@@ -55,37 +64,50 @@ namespace SettingsModelUnitTests
+                                              "guid" : "{6239A42C1DE449A380BDE8FDD045185C}"
+                                              })" };
+        const std::string profileWithRawGuid{ R"({
+                                              "name" : "profile4",
+                                              "guid" : "6239a42c-1de4-49a3-80bd-e8fdd045185c"
+                                              })" };
+        const std::string profileWithGuidFormatP{ R"({
+                                              "name" : "profile4",
+                                              "guid" : "(6239a42c-1de4-49a3-80bd-e8fdd045185c)\\"
+                                              })" };
+        // Valid GUIDs
         const std::string profileWithNullGuid{ R"({
                                               "name" : "profile3",
                                               "guid" : "{00000000-0000-0000-0000-000000000000}"
                                               })" };
-        const std::string profileWithGuid{ R"({
+        const std::string profileWithGuidFormatB{ R"({
                                               "name" : "profile4",
                                               "guid" : "{6239a42c-1de4-49a3-80bd-e8fdd045185c}"
                                               })" };
+        const std::string profileWithGuidUpperCaseFormatB{ R"({
+                                              "name" : "profile4",
+                                              "guid" : "{6239A42C-1DE4-49A3-80BD-E8FDD045185C}"
+                                              })" };
 
-        const auto profile0Json = VerifyParseSucceeded(profileWithoutGuid);
-        const auto profile1Json = VerifyParseSucceeded(secondProfileWithoutGuid);
-        const auto profile2Json = VerifyParseSucceeded(profileWithNullForGuid);
-        const auto profile3Json = VerifyParseSucceeded(profileWithNullGuid);
-        const auto profile4Json = VerifyParseSucceeded(profileWithGuid);
-
-        const auto profile0 = implementation::Profile::FromJson(profile0Json);
-        const auto profile1 = implementation::Profile::FromJson(profile1Json);
-        const auto profile2 = implementation::Profile::FromJson(profile2Json);
-        const auto profile3 = implementation::Profile::FromJson(profile3Json);
-        const auto profile4 = implementation::Profile::FromJson(profile4Json);
-        const winrt::guid cmdGuid = Utils::GuidFromString(L"{6239a42c-1de4-49a3-80bd-e8fdd045185c}");
-        const winrt::guid nullGuid{};
+        parseAndVerifyProfile(profileWithoutGuid, false);
+        parseAndVerifyProfile(secondProfileWithoutGuid, false);
 
-        VERIFY_IS_FALSE(profile0->HasGuid());
-        VERIFY_IS_FALSE(profile1->HasGuid());
-        VERIFY_IS_FALSE(profile2->HasGuid());
-        VERIFY_IS_TRUE(profile3->HasGuid());
-        VERIFY_IS_TRUE(profile4->HasGuid());
+        // The following crash JSON parsing

```

Comment:
FWIW, this should throw an exception which you can verify you caught with `VERIFY_THROWS`!

Reply:
I wasn't sure if they should crash so I left them as comment, now they are using VERIFY_THROWS (thanks for mentioning that!)